Fixing entropy calculation

abaheti95 opened the pull request:
I made a mistake when logging entropy by only looking at the actions rather than the entire distribution.
This fixes that mistake and now does correct entropy calculation

[mlflow of old and new entropy logging](https://dbc-559ffd80-2bfc.cloud.databricks.com/ml/experiments/1411130483624921?o=7395834863327820&searchFilter=&orderByKey=attributes.start_time&orderByAsc=false&startTime=ALL&lifecycleFilter=Active&modelVersionFilter=All+Runs&datasetsFilter=W10%3D&compareRunsMode=CHART)

## Update
I've modified the PR to include more functionalities:
1. Getting the sequence and token entropy of the sequences
2. seq entropy becomes the loss term
3. Token entropy is used to get different percentiles of token entropies throughout the batch

dakinggg reviewed (commented):
Since fixing a bug in a narrow, fairly self contained piece of the codebase, lets add unit tests while we're here?

gupta-abhay commented:
> Since fixing a bug in a narrow, fairly self contained piece of the codebase, lets add unit tests while we're here?

+1 to this

abaheti95 commented:
Okay... modified the code to calculate both token and sequence entropies and added unit tests for both.
Also added piping for entropy loss from @jdchang1 's [entropy loss branch](https://github.com/databricks/compose-rl/tree/jchang/entropy-grpo)

jdchang1 reviewed (approved):
LGTM! Thanks for the fix and tests

gupta-abhay reviewed (commented):
mostly lg, some nits!

abaheti95 commented on compose_rl/utils/utils.py:
```diff
@@ -135,23 +135,70 @@ def get_log_probs(
+        torch.Tensor: Entropy values for each item in the batch (batch_size,).
+    """
+    # Calculate per-token entropies
+    token_entropies = get_token_entropies(logits)
```
Thanks for the suggestion... I updated the `get_sequence_entropies` to now ingest `token_entropies` directly
